### ISSUE TITLE
add cgo.d.plugin

### ISSUE THREAD

**Issue by ilyam8** [netdata/netdata#5309]
##### Feature idea summary

Since we have some modules which uses `cgo` we need additional plugin.

The issue is opened in the netdata repo to track progress of v1.13-rc0.

##### Expected behavior

All go modules that uses `cgo` will be in `cgo.d.plugin`

**Comment by ilyam8** [comment]
This `cgo` plugin is quite problematic...

We can't have several `cgo` modules in one plugin, because they likely relies on different shared libraries and if some of them is not exist on target system the whole thing won't work.

It is worth mentioning that [`telegraf`](https://github.com/influxdata/telegraf) avoids using `cgo` and it has [tons of plugins](https://github.com/influxdata/telegraf/tree/master/plugins/inputs). So it's not mandatory at all.

Solutions:
 * separate plugin for every `cgo` module.

It will work, but imo **no go in general**. We can mantain some, but they won't be installed by default. This may be needed for specific requests from some [users](https://github.com/netdata/netdata/issues/5094) or something like that.

 * use `python`

Yes, yes, `python` is able to load libs dynamically. I think we should use `python` for modules which can't be written in `go` wo using `cgo`. Keep in mind - python is nice!

___

At the moment we have 2 cgo modules in go.d.plugin (they are disabled):
 - oracledb
 - nvidia_nvml

Not written, but to do:
- [Neo4j](https://github.com/netdata/netdata/pull/3709)

[Their go driver](https://github.com/neo4j/neo4j-go-driver#requirements-1) uses `cgo` too.


All of them can be rewritten in python, relevant python bindings:
 - [nvidia_nvml](https://pypi.org/project/nvidia-ml-py/) (official from NVIDIA)
 - [oracledb](https://pypi.org/project/cx-Oracle/)
 - [neo4j](https://github.com/neo4j/neo4j-python-driver)

___

tl;dr: implement modules in `python` if they require `cgo`. If, for some reason, it should be in `go`, make a separate plugin.

**Comment by ilyam8** [comment]
@gmosx @Ferroin @ktsaou @paulfantom @Wing924 @vlvkobal @kataras @cakrit

**Comment by ilyam8** [comment]
@netdata/all @Wing924

**Comment by ilyam8** [comment]
won't be added, at least for now i think we can't do it, if someone has other opinion feel free to comment
